packages/frontend-app-api/src/wiring/createSpecializedApp.tsx

```diff
@@ -95,21 +97,47 @@ function deduplicateFeatures(
 
 // Helps delay callers from reaching out to the API before the app tree has been materialized
 class AppTreeApiProxy implements AppTreeApi {
-  #safeToUse: boolean = false;
+  #routeInfo?: RouteInfo;
 
-  constructor(private readonly tree: AppTree) {}
+  constructor(
+    private readonly tree: AppTree,
+    private readonly appBasePath: string,
+  ) {}
 
-  getTree() {
-    if (!this.#safeToUse) {
+  private checkIfInitialized() {
+    if (!this.#routeInfo) {
       throw new Error(
         `You can't access the AppTreeApi during initialization of the app tree. Please move occurrences of this out of the initialization of the factory`,
       );
     }
+  }
+
+  getTree() {
+    this.checkIfInitialized();
+
     return { tree: this.tree };
   }
 
-  initialize() {
-    this.#safeToUse = true;
+  getNodesByRoutePath(sourcePath: string): { nodes: AppNode[] } {
+    this.checkIfInitialized();
+
+    let path = sourcePath;
+    if (sourcePath.startsWith(this.appBasePath)) {
```

Comment:
might as well use `path` here and below to shorten it a bit, I find it a bit more clear too